### PR TITLE
feat: add test for the section homepage

### LINKED ISSUE #1103: Create src/components/sections/homepage component test

### PR DESCRIPTION
**Pull request by Samantha-KY**
# Submit a pull request

- [ ] This is not a duplicate of an existing pull request.
- [ ] No existing features have been broken without good reason.
- [ ] Your commit messages are detailed
- [ ] The code style guideline have been followed.
- [ ] Documentation has been updated to reflect your changes.
- [ ] Tests have been added or updated to reflect your changes.
- [ ] All tests pass.

---

Replace any ":question:" below with information about your pull request.

## Pull Request Details

Closes #1103 

Provide details about your pull request and what it adds, fixes, or changes.

:question:

## Breaking Changes

Describe what features are broken by this pull request and why, if any.

:question:

## Issues Fixed

Enter the issue numbers resolved by this pull request below, if any.

1. :question:

## Other Relevant Information

Provide any other important details below.

### CODE REVIEW

File: __tests__/components/sections/homepage/OpenSource.test.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { screen } from "@testing-library/react";
+import { renderWithRedux } from "../../../../__mocks__/renderWithRedux";
+import OpenSource from "@/components/sections/homepage/OpenSource";
+
+describe("OpenSource", () => {
+  it("should render Open Source", () => {
+    renderWithRedux(<OpenSource />);
+    expect(screen.getByTestId("openSourceId")).toBeInTheDocument();
+    expect(screen.getByRole("link")).toBeInTheDocument();
+    expect(screen.getByRole("link").hasAttribute("href")).toBeTruthy();
```
Comment (mhenriette) [inline __tests__/components/sections/homepage/OpenSource.test.tsx:11]: avoid repetition by using link element variable 
```suggestion
     const link = screen.getByRole("link");
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "https://github.com/dacadeorg/dacade-frontend-app");
```

File: __tests__/components/sections/homepage/OpenSource.test.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    expect(screen.getByRole("link").getAttribute("href")).toBe("https://github.com/dacadeorg/dacade-frontend-app");
```
Comment (mhenriette) [inline __tests__/components/sections/homepage/OpenSource.test.tsx:12]: we can check if the link has the correct attribute `href` and right value in one go, look at the code i suggested above. hence, there's no need of this line

File: __tests__/components/sections/homepage/OpenSource.test.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  });
```
Comment (mhenriette) [inline __tests__/components/sections/homepage/OpenSource.test.tsx:13]: Add the test to check if the link renders the right text content

File: __tests__/components/sections/homepage/OpenSource.test.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+});
```
Comment (mhenriette) [inline __tests__/components/sections/homepage/OpenSource.test.tsx:14]: we should also test if the link click navigates to a different page

**Review by mhenriette — CHANGES_REQUESTED** [review]

File: __tests__/components/sections/homepage/OpenSource.test.tsx (hunk 4 above)
Comment (Samantha-KY) [reply __tests__/components/sections/homepage/OpenSource.test.tsx:14]: We can do this if we have an event being triggered

**Review by mbukeRepo — COMMENTED** [review]
Let's use Faker.js to generate some mock data for testing, 

https://www.npmjs.com/package/@faker-js/faker

**Review by mhenriette — APPROVED** [review]

**Review by serapieTuyishime — APPROVED** [review]

**Review by musayann — APPROVED** [review]

**Pull request merged**